File: node/src/examples/CounterAggregate.ts
import { AggregateRoot } from '../core/AggregateRoot'
import { Event } from '../core/Event'

export class CounterAggregate extends AggregateRoot {
  private count: number = 1

  get value(): number {
    return this.count
  }

  apply(event: Event): void {
      throw new Error('Method not implemented.')
  }

  increment(): void {
    throw new Error("Not supported!")
  }

  decrement(): void {
    throw new Error("Not supported!")
  }
}

File: dotnet/src/Examples/CounterAggregate.cs
using Core;

namespace Examples;

public record CounterIncremented();
public record CounterDecremented();

public partial class CounterAggregate : AggregateRoot
{
    public int Value { get; private set; }

    public CounterAggregate(string id)
        : base(id)
    {
    }

    public void Increment() {}
    public void Decrement() {}

    protected override void Apply(IEvent<object> @event)
    {
    }
}
